### DISCUSSION TITLE
关于api无法连接

### DISCUSSION THREAD

**Discussion by xhofe** [AlistGo/alist#4572] [Announcements]
由于原api `https://api.nn.ci`被sni阻断，导致国内无法连接的情况，故提供两个新的api供使用：

- `https://api-cf.nn.ci` (套了一层cloudflare)
- `https://api.xhofe.top` (辽宁多线， 由[亚洲云](https://www.asiayun.com/aff/QQCOOQKZ)赞助)

如何使用？

#### AList程序中

没记错的话，只有`aliyundrive_open`会使用到此api，故只需要将`aliyundrive_open`后台配置中的`Oauth令牌链接`更换为
- `https://api-cf.nn.ci/alist/ali_open/token`
- `https://api.xhofe.top/alist/ali_open/token`

#### AList文档中

在
![image](https://github.com/alist-org/alist/assets/36558727/a19bec06-b4f9-49f1-ad27-3e86ae90087e)
或
![image](https://github.com/alist-org/alist/assets/36558727/49a81931-1ca7-49c2-a2e1-28517baaec10)
切换为你可以使用的api。

**Comment by Whale778** [comment]
![image](https://github.com/alist-org/alist/assets/104715597/fd5a8f9b-7eb4-4e3d-9625-5735bb756e7c)
可是后台都进不去，怎么换那个qauth令牌链接呀

**Reply by xhofe** [reply]
**等**，因为api连不上，默认会重试三次+原本的一次，每次超时时间30s，即每个`aliyundrive_open`账号会尝试2分钟。

**Reply by Whale778** [reply]
刷新了是不是就要重新等了呀，我已经等了大概二十分钟了，期间等得不耐烦的时候就点刷新，目前是挂了四个阿里云盘加一个百度云

**Reply by lovezongzi** [reply]
在你的url后面加/@manage,或者/@manage/users

**Reply by naponorn** [reply]
都是大神啊

**Comment by anwen-anyi** [comment]
如果使用(天翼)Ocr接口 也需要修改一下 https://api.nn.ci/ocr/file/json

 - AList 管理 --> 设置 --> 全局 --> Ocr接口

**Reply by anwen-anyi** [reply]
### 阿里云盘open修改方式：
![image](https://github.com/alist-org/alist/assets/56105412/bc90dd81-35d5-4773-952c-d160830efce8)

**Reply by xxLIUJINHU** [reply]
oauth令牌在哪里

**Comment by zzerosama** [comment]
感谢大佬，改完Oauth令牌和token能用了，吓死我了，还以为被和谐了😭

**Comment by aioufu** [comment]
现在可以了感谢

**Comment by fioepq9** [comment]


**Reply by xhofe** [reply]
程序中没有
![image](https://github.com/alist-org/alist/assets/36558727/16de8483-f8ea-44a6-9923-bd7b8750faa1)

**Comment by ranran123987** [comment]
感谢大佬确实能用了

**Comment by Jasonanywhere** [comment]
我的直接在黑窗口处就连接不到
<img width="693" alt="image" src="https://github.com/alist-org/alist/assets/88193248/059a2ad5-dfa4-4662-a54a-9e6b046fa981">

**Reply by anwen-anyi** [reply]
没错啊 这里就是在加载 这里结束了就可以去AList后台改验证链接了

**Comment by Jasonanywhere** [comment]
可以了

**Comment by studyguy** [comment]
感谢大佬，完美解决

**Comment by liyialiyia** [comment]
完美解决，感谢感谢

**Comment by lazywen** [comment]
这么小众的东西居然被盯上了

**Comment by hoadi1230** [comment]
TokenVerifyFailed:The token was expected to have 3 parts, but got 1.

**Comment by btxl1021** [comment]
非常感谢，能用了

**Comment by BladPit** [comment]
效果立竿见影，非常棒，token也不用换，只换地址就马上可用了。这阿里真是一天天不停歇

**Comment by simear2004** [comment]
感谢大佬，改了，可以用 了

**Comment by linepro6** [comment]
似乎不是 sni 阻断，而是整个服务器的 TCP 443 端口被阻断了。
![Snipaste_2023-06-14_17-20-38](https://github.com/alist-org/alist/assets/53991395/e18e3147-4e31-4a62-b8cb-9a39471e0eec)
像是跑代理被识别了（

**Reply by xhofe** [reply]
但是我并没有在这台服务器上安装过任何代理相关的软件，全都是一些网站和后端服务。

有一个api是反向代理其他任意网站的，类似于https://www.j000e.com/cloudflare/cfworkers_reverse_proxy.html 可能是因为这个被墙了。

**Comment by MicroMatrixOrg** [comment]
阿里一天天的真不省事，这个东西能自建吗，要是能自建，我有家里的服务器还能加点东西，搭建下

**Reply by xiezhao** [reply]
我今天看过了，可以自建，阿里云盘有开发文档，但是需要懂开发

**Comment by itsHenry35** [comment]
> 似乎不是 sni 阻断，而是整个服务器的 TCP 443 端口被阻断了。 ![Snipaste_2023-06-14_17-20-38](https://user-images.githubusercontent.com/53991395/245742825-e18e3147-4e31-4a62-b8cb-9a39471e0eec.png) 像是跑代理被识别了（

正解
@Xhofe 我遇到过这种墙，通过换IP即可解决，virmach换IP貌似需要开工单后付5刀。推荐Microsoft Azure等云服务提供商，可以免费更换IP。

**Comment by lilangus181** [comment]
感谢大佬！解决了

**Comment by html5syt** [comment]
@Xhofe 请问搭建在国外的alist，在国内访问，要换api吗

**Reply by Sunbangyan233** [reply]
服务端能连接到就行，试过不用换。国内的需要换。

**Comment by alge-cn** [comment]
/@manage进不去后台呢 ....是需要升级最新的docker镜像吗？

**Comment by ghost** [comment]
请问这怎么办

Failed init storage but storage is already created: failed init storage: TokenVerifyFailed:The token was expected to have 3 parts, but got 1.

**Comment by zhlx2835** [comment]
提示这个咋办：failed get objs: failed to list objs: NotFound.FileId:The resource file_id cannot be found. file_id:

**Reply by haodayitiaolong** [reply]
兄弟解决了吗？

**Comment by cubricmms** [comment]
> 似乎不是 sni 阻断，而是整个服务器的 TCP 443 端口被阻断了。 ![Snipaste_2023-06-14_17-20-38](https://user-images.githubusercontent.com/53991395/245742825-e18e3147-4e31-4a62-b8cb-9a39471e0eec.png) 像是跑代理被识别了（

叉个题外话，这个是什么工具呀

**Reply by littleyz** [reply]
ping.pe

**Comment by xxLIUJINHU** [comment]
Oauth令牌链接z在哪里填

**Reply by MicroMatrixOrg** [reply]
去alist官网那里，有设置存储的地方，找阿里云盘

**Comment by likevi9528** [comment]
https://api-cf.nn.ci/alist/ali_open/token
这个api域名解析失败。
ping api-cf.nn.ci
ping: api-cf.nn.ci: Temporary failure in name resolution
![image](https://github.com/alist-org/alist/assets/81839775/8bcc78dc-fb02-4eb6-903f-304b5781292e)

**Comment by YQRYYDS** [comment]
这是什么问题？？
Failed init storage: Post "https://openapi.aliyundrive.com/adrive/v1.0/user/getDriveInfo": dial tcp ************53:443: connect: connection refused

**Reply by wangyilin188** [reply]
Oauth令牌，隔段时间就不能用了，要换成另一个就好了，但过一段时间此Oauth 又不行了，换回原来的又可以了。

**Reply by winter-rabbit** [reply]
是的，隔段时间不能用，只需到管理里点住阿里云盘 的disable,然后enable,就又行了，不用换Oauth

**Comment by 0x000-0** [comment]
Post "https://openapi.alipan.com/adrive/v1.0/openFile/getUploadUrl": net/http: TLS handshake timeout

境外服务器经常 连不上，改了hosts 指定ipv6 好了点，时间长了，有时候还是会超时，能不能把这个api也套一层代理

**Comment by xingxinag** [comment]
我终于能扫码了

**Comment by Jasonsun77** [comment]
今天又不行了，Oauth令牌链接更换为以下两个也不行
https://api-cf.nn.ci/alist/ali_open/token
https://api.xhofe.top/alist/ali_open/token

**Reply by AxisRay** [reply]
<img width="1051" alt="image" src="https://github.com/user-attachments/assets/a3446a49-e32e-45cd-80e1-eab7ed3a3982">
看起来挂了

**Reply by Jasonsun77** [reply]
现在好了，应该是修复了

**Reply by AspirantH** [reply]
又不行了

**Comment by baygixu** [comment]
请问怎么进入修改Oauth令牌啊？谢谢

**Reply by dengshenkk** [reply]
存储/阿里云盘/编辑/Oauth

**Comment by LikeLucia** [comment]


**Reply by denki-san** [reply]
是这样的

**Comment by LocSiege** [comment]
failed to refresh token: refresh token is empty, resp: 404 page not found换token也还是失效怎么办

**Comment by Liu0345** [comment]
![PixPin_2025-03-17_20-21-31](https://github.com/user-attachments/assets/c91c8e88-838e-4fef-9965-cb045173e26c)
换了还是不行，是不是又屏蔽了，有没有好的方法

**Comment by allindream** [comment]
请问出现这种是什么问题，一直不知道问题出现在哪里，求指点？
![image](https://github.com/user-attachments/assets/ead9d4a7-1a64-42c0-9bde-46d8d113480f)
![image](https://github.com/user-attachments/assets/5f9c9e88-1fc1-46de-8cfb-1ad26f2c71a3)

**Comment by allindream** [comment]
再补一张图吧，扫码获取token,总是报错，Error  InvalidClientSecret: invalid client_secret，不知道什么原因，怎么改。
![image](https://github.com/user-attachments/assets/284b5b84-4b99-4123-8333-2d8f418dc6b5)

**Reply by html5syt** [reply]
由于一些不好的问题，原作者已经停止了部分API服务，建议去OpenList，他们有提供最新的API地址和服务。

**Reply by liu-ce** [reply]
OpenList提供的API地址可以放Alist里使用不

**Reply by html5syt** [reply]
不行，因为openlist对api进行了一些修改以适合自建授权api的用户使用。

**Comment by yppbrid** [comment]
同楼上问题一样

**Comment by zengdw** [comment]
保存报错了
<img width="1705" height="594" alt="1767791000792" src="https://github.com/user-attachments/assets/63254cba-9752-43c6-840e-09994ff17d84" />
